plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -69,96 +84,269 @@ protected DistributedQueryRunner createQueryRunner()
-        private Predicate<String> pathPredicate = path -> true;
-        private Optional<TrackingFileIoProvider.OperationType> type = Optional.empty();
+        private final FileType fileType;
+        private final OperationType operationType;
 
-        public OperationCounts(Map<OperationContext, Integer> allCounts)
+        public FileOperation(OperationContext operationContext)
         {
-            this.allCounts = requireNonNull(allCounts, "allCounts is null");
+            this(fromFilePath(operationContext.getFilePath()), operationContext.getOperationType());
         }
 
-        public OperationCounts forQueryId(String queryId)
+        public FileOperation(FileType fileType, OperationType operationType)
         {
-            this.queryId = Optional.of(queryId);
-            return this;
+            this.fileType = requireNonNull(fileType, "fileType is null");
+            this.operationType = requireNonNull(operationType, "operationType is null");
         }
 
-        public OperationCounts forPathContaining(String value)
+        @Override
+        public boolean equals(Object o)
         {
-            this.pathPredicate = path -> path.contains(value);
-            return this;
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            FileOperation that = (FileOperation) o;
+            return fileType == that.fileType && operationType == that.operationType;
         }
 
-        public OperationCounts forOperation(TrackingFileIoProvider.OperationType type)
+        @Override
+        public int hashCode()
         {
-            this.type = Optional.of(type);
-            return this;
+            return Objects.hash(fileType, operationType);
         }
 
-        public Map<OperationContext, Integer> get()
+        @Override
+        public String toString()
         {
-            return allCounts.entrySet().stream()
-                    .filter(entry -> queryId.map(id -> entry.getKey().getQueryId().equals(id)).orElse(true))
-                    .filter(entry -> pathPredicate.test(entry.getKey().getFilePath()))
-                    .filter(entry -> type.map(value -> value == entry.getKey().getOperationType()).orElse(true))
-                    .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+            return toStringHelper(this)
+                    .add("fileType", fileType)
+                    .add("operationType", operationType)
+                    .toString();
         }
+    }
+
+    enum FileType
+    {
+        METADATA_JSON,
+        MANIFEST,
+        SNAPSHOT,
+        /**/;
 
-        public int sum()
+        public static FileType fromFilePath(String path)
         {
-            return get().values().stream().mapToInt(Integer::intValue).sum();
+            if (path.contains("metadata.json")) {
```

Comment:
These should really be comparing just the name portion and not the directory. If we had a path like `/data/metadata.json/abc` then this should not match.

Reply:
yeah I see your point but in the second case with '/snap-' it is at the begining of the file name and we cannot really make any assumptions about other parts. What is more my understanding is that these tests don't test for validity but only count number of files accesses and for this I think they are enough. 
Anyway I modified conditions for metadata and manifest.